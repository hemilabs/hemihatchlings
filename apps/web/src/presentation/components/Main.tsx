import React from 'react'

function Main(): JSX.Element {
  return (
    <h1 className='text-2xl'>Home Page</h1>
  )
}

export default Main
