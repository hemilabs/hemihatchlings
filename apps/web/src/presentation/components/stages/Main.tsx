import { useContext, useEffect, useState } from 'react'
import { servicesContext } from '../../context'
import ProgressBar from './ProgressBar'
import OptInButton from './OptInButton'
import Eggs from './Eggs'
import Hatchling from './Hatchling'
import StatusMessage from './StatusMessage'
import { ElementEnum, Stage, StageEnum } from '@hemihatchlings/shared'

function Main(): JSX.Element {
  const [stage, setStage] = useState<StageEnum>(StageEnum.Egg)
  const [address, setAddress] = useState('')
  const [transactionHash, setTransactionHash] = useState('')
  const [element, setElement] = useState<ElementEnum>(ElementEnum.Fire)

  const {
    connectWalletService,
    mintNFTService,
    getHatchlingStageService
  } = useContext(servicesContext)

  const connectWallet = async () => {
    const response = await connectWalletService.execute()

    setAddress(response.address)
  }

  const mintNFT = async () => {
    const response = await mintNFTService.execute()

    setTransactionHash(response.transactionHash)
  }

  const hatchEgg = async (element: ElementEnum) => {
    await connectWallet()
    await mintNFT()

    setElement(element)
    setStage(StageEnum.Baby)
  }

  const fetchHatchlingStage = async () => {
    const response = await getHatchlingStageService.execute({ transactionHash })
    const stageValue = response?.stage as keyof typeof StageEnum
    
    setStage(StageEnum[stageValue])
  }

  useEffect(() => {
    if (stage !== StageEnum.Egg) {
      fetchHatchlingStage()
    }
  }, [])

  return (
  <>
    <div className={stage === StageEnum.Egg ? '' : 'hidden'}>
      <Eggs hatchEgg={hatchEgg}/>
    </div>

    <div className={stage != StageEnum.Egg ? '' : 'hidden'}>
      <Hatchling element={element} stage={stage}/>

      <div className="mt-4 sm:mt-8 lg:mt-12">
        <OptInButton />
      </div>
      <div className="mt-4 sm:mt-8 lg:mt-12">
        <ProgressBar percent={8}/>
        {address}
        <StatusMessage stage={stage} />
      </div>
    </div>
  </>
  )
}

export default Main
