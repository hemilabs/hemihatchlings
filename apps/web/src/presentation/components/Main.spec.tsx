import { describe, it, expect} from 'vitest'
import React from 'react'
import { render } from '@testing-library/react'
import Main from './Main'

describe('src/presentation/Main', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<Main />)

    expect(asFragment()).toMatchSnapshot()
  })
})
