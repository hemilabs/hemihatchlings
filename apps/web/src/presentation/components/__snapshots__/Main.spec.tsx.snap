// Vitest Snapshot v1

exports[`src/presentation/Main > renders correctly 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col container"
  >
    <header
      class="flex flex-row w-full justify-center items-center h-24 gap-x-8 sm:h-28 lg:h-32"
    >
      <div>
        <h1
          class="font-poller text-white text-lg sm:text-xl lg:text-2xl"
        >
          BB
        </h1>
      </div>
      <div
        class="grow border h-0 border-b-0 max-h-0 max-w-5xl w-auto border-white"
      />
      <div>
        <ul
          class="font-monaco text-white text-sm text-right sm:text-lg lg:text-xl"
        >
          <a
            class="hover:text-gray-400"
            href="#"
          >
            <li>
              GITHUB
            </li>
          </a>
          <a
            class="hover:text-gray-400"
            href="#"
          >
            <li>
              COMMUNITY
            </li>
          </a>
          <a
            class="hover:text-gray-400"
            href="#"
          >
            <li>
              TWITTER
            </li>
          </a>
        </ul>
      </div>
    </header>
    <div
      class="mt-4 sm:mt-8 lg:mt-12"
    >
      <div
        class="flex flex-col justify-center items-center h-24"
      >
        <img
          alt="logo"
          class=""
          src="/src/assets/logo.svg"
        />
        <h2
          class="font-poller text-white text-center text-xs mt-4 sm:text-lg md:text-xl lg:text-3xl"
        >
          Learn how Ethereum & Bitcoin Finality Works.
        </h2>
      </div>
    </div>
    <div
      class="mt-4 sm:mt-8 lg:mt-20"
    >
      <div
        class="flex flex-row justify-center gap-4 lg:gap-12"
      >
        <button
          class="disabled:opacity-50 lg:hidden"
          disabled=""
        >
          <img
            alt="Go Left"
            class="w-8 rotate-180"
            src="/src/assets/chevron.svg"
          />
        </button>
        <div
          class=""
        >
          <button>
            <div
              class="p-8 border border-white bg-fire-yellow"
            >
              <img
                alt="Fire Egg"
                src="/src/assets/fire-egg.png"
              />
            </div>
            <div
              class="font-monaco text-sm text-white mt-2 text-left sm:text-base md:text-lg md:mt-4"
            >
              01
            </div>
          </button>
        </div>
        <div
          class="hidden lg:block"
        >
          <button>
            <div
              class="p-8 border border-white bg-water-blue"
            >
              <img
                alt="Water Egg"
                src="/src/assets/water-egg.png"
              />
            </div>
            <div
              class="font-monaco text-sm text-white mt-2 text-left sm:text-base md:text-lg md:mt-4"
            >
              02
            </div>
          </button>
        </div>
        <div
          class="hidden lg:block"
        >
          <button>
            <div
              class="p-8 border border-white bg-grass-green"
            >
              <img
                alt="Grass Egg"
                src="/src/assets/grass-egg.png"
              />
            </div>
            <div
              class="font-monaco text-sm text-white mt-2 text-left sm:text-base md:text-lg md:mt-4"
            >
              03
            </div>
          </button>
        </div>
        <button
          class="disabled:opacity-50 lg:hidden"
        >
          <img
            alt="Go Right"
            class="w-8"
            src="/src/assets/chevron.svg"
          />
        </button>
      </div>
      <div
        class="flex flex-col items-center"
      >
        <h2
          class="font-monaco font-normal text-white text-lg mt-8 sm:text-xl md:text-2xl lg:text-3xl lg:mt-12 xl:text-4xl"
        >
          FREE MINT + GAS FEES
        </h2>
      </div>
    </div>
    <div
      class="mt-8 sm:mt-8 mb-8 lg:mt-16"
    >
      <div
        class="flex flex-col items-center gap-8"
      >
        <div
          class="flex flex-col max-w-screen-xl items-center p-8 border border-white"
        >
          <div
            class="flex flex-row justify-between w-full"
          >
            <h2
              class="font-space font-bold text-white text-lg hover:text-gray-400 sm:text-xl md:text-2xl lg:text-3xl xl:text-4xl"
            >
              HOW IT WORKS
            </h2>
            <img
              alt="Minus Icon"
              class="w-4 md:w-6 lg:w-8"
              src="/src/assets/minus-icon.png"
            />
          </div>
          <div
            class="font-space font-light text-white text-sm mt-4 sm:text-base md:text-lg lg:text-xl"
          >
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Vel fringilla est ullamcorper eget nulla facilisi.
            </p>
            <br />
            <p>
              Ut aliquam purus sit amet luctus venenatis. Egestas sed sed risus pretium quam vulputate dignissim. Adipiscing at in tellus integer feugiat. Pulvinar proin gravida hendrerit lectus.
            </p>
            <br />
            <p>
              Eleifend donec pretium vulputate sapien nec sagittis aliquam. Donec ac odio tempor.
            </p>
          </div>
        </div>
        <div
          class="flex flex-row justify-between w-full max-w-screen-xl items-center p-8 border border-white"
        >
          <h2
            class="font-space font-normal text-white text-lg sm:text-xl md:text-2xl lg:text-3xl xl:text-4xl"
          >
            ADD A FEATURE
          </h2>
          <img
            alt="Plus Icon"
            class="w-4 md:w-6 lg:w-8"
            src="/src/assets/plus-icon.png"
          />
        </div>
        <div
          class="flex flex-row justify-between w-full max-w-screen-xl items-center p-8 border border-white"
        >
          <h2
            class="font-space font-normal text-white text-lg sm:text-xl md:text-2xl lg:text-3xl xl:text-4xl"
          >
            COMMUNITY
          </h2>
          <img
            alt="Plus Icon"
            class="w-4 md:w-6 lg:w-8"
            src="/src/assets/plus-icon.png"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
